conans/model/graph_lock.py

```diff
@@ -494,7 +494,6 @@ def pre_lock_node(self, node):
             node.graph_lock_node = locked_node
             if locked_node.options is not None:  # This was a "partial" one, not a "base" one
                 node.conanfile.options.values = locked_node.options
-                node.conanfile.options.freeze()
```

Comment:
This is the key of this PR: the options are not freezed anymore, so they do not error while being evaluated and getting different values.